ApplicationLibCode/ProjectDataModel/RimEclipseResultDefinitionTools.cpp

```diff
@@ -0,0 +1,770 @@
+    {
+        flowResultsData->minMaxScalarValues( resAddr, timeStep, &localMin, &localMax );
+        flowResultsData->posNegClosestToZero( resAddr, timeStep, &localPosClosestToZero, &localNegClosestToZero );
+    }
+    else
+    {
+        localMin = globalMin;
+        localMax = globalMax;
+
+        localPosClosestToZero = globalPosClosestToZero;
+        localNegClosestToZero = globalNegClosestToZero;
+    }
+
+    CVF_ASSERT( legendConfigToUpdate );
+
+    legendConfigToUpdate->disableAllTimeStepsRange( true );
+    legendConfigToUpdate->setClosestToZeroValues( globalPosClosestToZero, globalNegClosestToZero, localPosClosestToZero, localNegClosestToZero );
+    legendConfigToUpdate->setAutomaticRanges( globalMin, globalMax, localMin, localMax );
+
+    if ( resultDefinition->hasCategoryResult() )
+    {
+        RimEclipseView* eclView = nullptr;
+        resultDefinition->firstAncestorOrThisOfType( eclView );
+
+        if ( eclView )
+        {
+            std::set<std::tuple<QString, int, cvf::Color3ub>, TupleCompare> categories;
+
+            std::vector<QString> tracerNames = flowDiagSolution->tracerNames();
+            int                  tracerIndex = 0;
+
+            for ( const auto& tracerName : tracerNames )
+            {
+                cvf::Color3ub color( cvf::Color3::GRAY );
+
+                RimSimWellInView* well = eclView->wellCollection()->findWell( RimFlowDiagSolution::removeCrossFlowEnding( tracerName ) );
+
+                if ( well ) color = cvf::Color3ub( well->wellPipeColor() );
+
+                categories.insert( std::make_tuple( tracerName, tracerIndex, color ) );
+                ++tracerIndex;
+            }
+
+            std::vector<std::tuple<QString, int, cvf::Color3ub>> categoryVector;
+
+            if ( resultDefinition->showOnlyVisibleCategoriesInLegend() )
+            {
+                std::set<int> visibleTracers =
+                    RigVisibleCategoriesCalculator::visibleFlowDiagCategories( *eclView, *flowResultsData, resAddr, timeStep );
+                for ( auto tupIt : categories )
+                {
+                    int tracerIndex = std::get<1>( tupIt );
+                    if ( visibleTracers.count( tracerIndex ) ) categoryVector.push_back( tupIt );
+                }
+            }
+            else
+            {
+                for ( const auto& tupIt : categories )
+                {
+                    categoryVector.push_back( tupIt );
+                }
+            }
+
+            legendConfigToUpdate->setCategoryItems( categoryVector );
+        }
+    }
+}
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+void RimEclipseResultDefinitionTools::updateCellResultLegend( RD* const resultDefinition, RimRegularLegendConfig* legendConfigToUpdate, int timeStep )
```

Comment:
RD? Use full class name.

Reply:
Added `using RD` to make the header file more readable. Use full class name in cpp.